### ISSUE TITLE
Seems no lock

### ISSUE THREAD

**Issue by erickuo13** [anand-seeenivasagam/mongoose-transaction#2]
Hi , I am very like this libs , specially using mongoose , it's great , but this libs seems not lock doc when processing transaction , would you have any concerns ? because transaction not only need handle rollback but also handle doc(Row) lock , do you have any idea ?

**Comment by michael-r-zhang** [comment]
I agree with you. Maybe, there's something wrong when the lib handles many transactions concurrently.

**Comment by lrcry** [comment]
I think we can use some similar mechanics like optimistic lock or MVCC to solve the concurrent lock issue i.e. to give an extra version number (or use the `__v` in mongoose models), to be used in update operations. I don't think it is within the scope of this simple library. @erickuofucker @michael-r-zhang
